framework/integration/wrapper/spec.go

```diff
@@ -0,0 +1,20 @@
+// +build k8srequired
+
+package wrapper
+
+import "github.com/giantswarm/operatorkit/framework"
+
+type Interface interface {
+	// CRUD functions for objects.
+	CreateObject(namespace string, obj interface{}) (interface{}, error)
+	DeleteObject(name, namespace string) error
+	GetObject(name, namespace string) (interface{}, error)
+	UpdateObject(namespace string, obj interface{}) (interface{}, error)
```

Comment:
I don't think doesn't make sense here. If you know what you have to pass, I believe you also know what concrete implementation you should use.

Reply:
I don't follow, the idea here is to pass different objects (CRs or standard K8s objects) and have the same test code for both. How do I know which implementation to use?

Reply:
How do you know which object to create?

Reply:
When you switch to table-driven test that can be tricky, I admit.

Reply:
The object can just be part of the data inside the table test config, same with a small function for the setup of the wrapper. Then I can use the interface throughout the whole test and we have the same code for each object. 

I don't understand how I could get this properly typed without basically writing the whole test twice.

Reply:
Yes, with table-dirven tests you're right.